acai-ios/CheckoutViewController.swift

```diff
@@ -0,0 +1,179 @@
+        tipAmountLabel.textColor = .black
+        view.addSubview(tipAmountLabel)
+        tipAmountLabel.snp.makeConstraints { (make) in
+            make.trailing.equalToSuperview().offset(-18)
+            make.bottom.equalTo(taxAmountLabel.snp.top)
+        }
+        subTotalLabel = UILabel()
+        subTotalLabel.translatesAutoresizingMaskIntoConstraints = false
+        subTotalLabel.text = "Subtotal: "
+        subTotalLabel.font = Acai.avenirNextMedium.withSize(20)
+        subTotalLabel.textColor = .black
+        view.addSubview(subTotalLabel)
+        subTotalLabel.snp.makeConstraints { (make) in
+            make.leading.equalToSuperview().offset(18)
+            make.bottom.equalTo(tipLabel.snp.top)
+        }
+        subTotalAmountLabel = UILabel()
+        subTotalAmountLabel.translatesAutoresizingMaskIntoConstraints = false
+        subTotalAmountLabel.text = "$subtotal"
+        subTotalAmountLabel.font = Acai.avenirNextMedium.withSize(20)
+        subTotalAmountLabel.textColor = .black
+        view.addSubview(subTotalAmountLabel)
+        subTotalAmountLabel.snp.makeConstraints { (make) in
+            make.trailing.equalToSuperview().offset(-18)
+            make.bottom.equalTo(tipAmountLabel.snp.top)
+        }
+        
+        let layout = UICollectionViewFlowLayout()
+        layout.scrollDirection = .vertical
+        
+        collectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        collectionView.backgroundColor = .white
+        collectionView.showsVerticalScrollIndicator = false
+        view.addSubview(collectionView)
+        collectionView.snp.makeConstraints { (make) in
+            make.top.equalTo(view.safeAreaLayoutGuide.snp.top)
+            make.bottom.equalTo(subTotalLabel.snp.top)
+            make.leading.equalToSuperview()
+            make.trailing.equalToSuperview()
+        }
+        
+        collectionViewListAdapter = ListAdapter(updater: ListAdapterUpdater(), viewController: self, workingRangeSize: 0)
+        collectionViewListAdapter.collectionView = collectionView
+        collectionViewListAdapter.dataSource = self
+        
+    }
+    
+    @objc func checkoutItems() {
+        print("checkoutButton clicked")
+    }
+
+}
+
+extension CheckoutViewController: ListAdapterDataSource {
+    func objects(for listAdapter: ListAdapter) -> [ListDiffable] {
+        
+        //hard code
+        checkoutMenuItems = [Acai.miamiBowl, Acai.testBowl]
+        let list = [Acai.menuItemCheckoutListIdentifier]
+        
+        return list as [ListDiffable]
+    }
+    
+    func listAdapter(_ listAdapter: ListAdapter, sectionControllerFor object: Any) -> ListSectionController {
+        if let object = object as? String, object == Acai.menuItemCheckoutListIdentifier {
+            let checkoutMenuItemSectionController = CheckoutMenuItemSectionController()
```

Comment:
If you return a list of `MenuItem` objects above, then you can do `if let object = object as? MenuItem` and then pass in that single object - versus the entire list of `MenuItem` objects - into the section controller. Your section controller controls a single `MenuItem`, so you should initialize it with a single `MenuItem` instead of the entire list.